django_statsd/tests.py

```diff
@@ -59,222 +61,235 @@
     def test_toolbar_timing(self):
-        client = get_client()
-        self.assertEqual(client.timings, [])
-        client.timing('testing', 1)
-        self.assertEqual(client.timings[0][0], 'testing|timing')
-        self.assertEqual(client.timings[0][2], 1)
+        statsd = StatsdClientProxy()
 
-    @patch.object(settings, 'STATSD_CLIENT', 'django_statsd.clients.toolbar')
+        self.assertEqual(statsd.timings, [])
+        statsd.timing('testing', 1)
+        self.assertEqual(statsd.timings[0][0], 'testing|timing')
+        self.assertEqual(statsd.timings[0][2], 1)
+
+    @override_settings(STATSD_CLIENT='django_statsd.clients.toolbar')
     def test_toolbar_gauge(self):
-        client = get_client()
-        self.assertEqual(client.cache, {})
-        client.gauge('testing', 1)
-        self.assertEqual(client.cache, {'testing|gauge': [[1, 1]]})
-        client.gauge('testing', 1, delta=True)
-        self.assertEqual(client.cache, {'testing|gauge': [[1, 1], [1, 1]]})
-
-    @patch.object(settings, 'STATSD_CLIENT', 'django_statsd.clients.toolbar')
+        statsd = StatsdClientProxy()
+
+        self.assertEqual(statsd.cache, {})
+        statsd.gauge('testing', 1)
+        self.assertEqual(statsd.cache, {'testing|gauge': [[1, 1]]})
+        statsd.gauge('testing', 1, delta=True)
+        self.assertEqual(statsd.cache, {'testing|gauge': [[1, 1], [1, 1]]})
+
+    @override_settings(STATSD_CLIENT='django_statsd.clients.toolbar')
     def test_toolbar_set(self):
-        client = get_client()
-        self.assertEqual(client.cache, {})
-        client.set('testing', 1)
-        self.assertEqual(client.cache, {'testing|set': [[1, 1]]})
+        statsd = StatsdClientProxy()
+
+        self.assertEqual(statsd.cache, {})
+        statsd.set('testing', 1)
+        self.assertEqual(statsd.cache, {'testing|set': [[1, 1]]})
 
-    @patch.object(settings, 'STATSD_CLIENT', 'django_statsd.clients.log')
+    @override_settings(STATSD_CLIENT='django_statsd.clients.log')
     @log_capture()
     def test_log_client(self, l):
-        client = get_client()
-        client.timing('testing.timing', 1)
-        client.incr('testing.incr')
-        client.decr('testing.decr')
-        client.gauge('testing.gauge', 1)
+        statsd = StatsdClientProxy()
+
+        statsd.timing('testing.timing', 1)
+        statsd.incr('testing.incr')
+        statsd.decr('testing.decr')
+        statsd.gauge('testing.gauge', 1)
         l.check(('statsd', 'INFO', 'Timing: testing.timing, 1, 1'),
                 ('statsd', 'INFO', 'Increment: testing.incr, 1, 1'),
                 ('statsd', 'INFO', 'Decrement: testing.decr, 1, 1'),
                 ('statsd', 'INFO', 'Gauge: testing.gauge, 1, 1'))
 
+    @override_settings(STATSD_CLIENT='django_statsd.clients.toolbar', STATSD_REFRESH_SECONDS=1)
+    def test_refresh_client(self):
+        statsd = StatsdClientProxy()
+        statsd.incr('test')
+        old_client = statsd._client
+        sleep(1)
```

Comment:
Maybe it's better to set the refresh timer at 2 so you can also test if not new client is instantiated before the timer has passed?

Reply:
Fair enough. Amended.